src/main/webapp/js/feed-loader.js

```diff
@@ -19,28 +19,89 @@ function fetchMessages(){
   });
 }
 
-function buildMessageDiv(message){
+/** Fetches replies and adds them to their parent message. */
+function fetchReplies(message) {
+  const replyThread = document.createElement('div');
+  replyThread.classList.add('reply-thread');
+
+  const url = '/messages?parent=' + message.id.toString();
+  fetch(url)
+      .then((response) => {
+        return response.json();
+      })
+      .then((messages) => {
+        messages.forEach((reply) => {
+          const replyDiv = buildMessageDiv(reply);
+          replyThread.appendChild(replyDiv);
+        });
+      });
+  return replyThread;
+}
+
+/**
+ * Builds an element that displays the message.
+ * @param {Message} message
+ * @return {Element}
+ */
+function buildMessageDiv(message) {
   const headerDiv = document.createElement('div');
   headerDiv.classList.add('message-header');
   headerDiv.appendChild(document.createTextNode(
-      message.user + ' - ' + 
+      message.user + ' - ' +
       new Date(message.timestamp) +
       ' [' + message.sentimentScore + ']'));
-  
 
   const bodyDiv = document.createElement('div');
   bodyDiv.classList.add('message-body');
   bodyDiv.innerHTML = message.text;
-  
 
   const messageDiv = document.createElement('div');
-  messageDiv.classList.add("message-div");
+  messageDiv.classList.add('message-div');
   messageDiv.appendChild(headerDiv);
   messageDiv.appendChild(bodyDiv);
 
+  fetch('/login-status')
+      .then((response) => {
+        return response.json();
+      })
+      .then((loginStatus) => {
+        if (loginStatus.isLoggedIn) {
+          const replyForm = buildReplyForm(message);
+          messageDiv.appendChild(replyForm);
+        }
+
+        const replyThread = fetchReplies(message);
+        messageDiv.appendChild(replyThread);
+      });
+
   return messageDiv;
 }
 
+/**
+ * Builds a reply form for the message.
+ * @param {Message} message
+ * @return {Element}
+ */
+function buildReplyForm(message) {
+  const textArea = document.createElement('textarea');
+  textArea.name = 'text';
+
+  const linebreak = document.createElement('br');
+
+  const input = document.createElement('input');
+  input.type = 'submit';
+  input.value = 'Submit';
+
+  const replyForm = document.createElement('form');
+  replyForm.action = '/messages?parent=' + message.id.toString();
+  replyForm.method = 'POST';
+  replyForm.appendChild(textArea);
+  replyForm.appendChild(linebreak);
+  replyForm.appendChild(input);
+
+  return replyForm;
+}
+
```

Comment:
Is there any particular reason this needs to be a build function rather than an element that you change to not be hidden if the user is logged in, and set the action of? It seems like it would be way more intensive to build after every single message when almost all the parts of it are constants and it's just its presence and its action link that's variable.

Reply:
Even though, the form itself is constant, the instances of it are variable due to the variable number of messages we have to load. Since we're appending the reply forms to each message, it is part of the building process and we can't really toggle them to hidden until after they are individually built.